hw_25/resources/pages/contactlist_page.py

```diff
@@ -0,0 +1,66 @@
+"""Contact List page"""
+
+
+import time
+from selenium.webdriver.common.by import By
+from selenium.webdriver.support.ui import WebDriverWait
+from selenium.webdriver.support import expected_conditions as EC
+from hw_25.resources.pages.base_page import BasePage
+from hw_25.resources.locators.add_contact_locators import AddContactLocators
+from hw_25.resources.locators.update_contact_locators import UpdateContactLocators
+from hw_25.resources.locators.delete_contact_locators import DeleteContactLocators
+
+
+class ContactListPage(BasePage):
+    def add_new_contact(self, contact_data):
+        self.driver.find_element(By.CSS_SELECTOR, AddContactLocators.add_new_contact_button).click()
+        self._fill_contact_form(contact_data)
+        self.driver.find_element(By.CSS_SELECTOR, AddContactLocators.submit_button_add_contact).click()
+
+    def update_contact(self, contact_data):
+        contacts = self.driver.find_elements(By.CSS_SELECTOR, UpdateContactLocators.contact_table_body_row_edit)
+        if contacts:
+            contacts[0].click()  # Выберите контакт для редактирования
+            self.driver.find_element(By.CSS_SELECTOR, UpdateContactLocators.edit_contact_button).click()
+            self._fill_contact_form(contact_data, is_update=True)
+            submit_button = self.driver.find_element(By.CSS_SELECTOR, UpdateContactLocators.submit_button_edit_contact)
+            submit_button.click()
+            return_to_contact_list_button = self.driver.find_element(
+                By.CSS_SELECTOR, UpdateContactLocators.return_to_contact_list_button)
+            return_to_contact_list_button.click()
+        else:
+            raise Exception("No contacts available to update.")
+
+    def delete_contact(self):
+        contacts = self.driver.find_elements(By.CSS_SELECTOR, DeleteContactLocators.contact_table_body_row_delete)
+        if contacts:
+            contacts[0].click()
+            self.driver.find_element(By.CSS_SELECTOR, DeleteContactLocators.delete_contact_button).click()
+            WebDriverWait(self.driver, 10).until(EC.alert_is_present())
+            self.driver.switch_to.alert.accept()
+        else:
+            raise Exception("No contacts available to delete.")
+
+    def _fill_contact_form(self, contact_data, is_update=False):
+        locators = UpdateContactLocators if is_update else AddContactLocators
+        wait = WebDriverWait(self.driver, 10)
+
+        def fill_field(locator, value):
+            element = wait.until(EC.element_to_be_clickable((By.CSS_SELECTOR, locator)))
+            element.clear()
+            element.send_keys(value)
+
+        time.sleep(3)
```

Comment:
you are using 2 different types of waiting and still use `sleep`?

Reply:
I use `time.sleep` here because with any other waiting **Edit contact page** does not have time to clear and fill the fields with new data. I tried `implicitly_wait` and `WebDriverWait`, only `time.sleep` works correctly.

Reply:
I've added self.find_element_visible, but it still is not working without time.sleep )) 
proof in attach 

https://github.com/user-attachments/assets/f89185c2-af46-4aed-8369-d1d409915a62

Reply:
Really strange, the data appears in the element with a small delay.
You can try to wait to page load:
```
    def wait_page_to_load(self):
        WebDriverWait(self.driver, 10).until(
            lambda driver: driver.execute_script('return document.readyState') == 'complete')
```

Reply:
i'll try! thank you)